frontend/public/components/namespace.jsx

```diff
@@ -534,127 +534,122 @@ const projectHeaderWithoutActions = () =>
 
 const projectRowStateToProps = ({ UI }) => ({
   metrics: UI.getIn(['metrics', 'namespace']),
-  selectedColumns: UI.getIn(['columnManagement']),
 });
 
 const ProjectTableRow = connect(projectRowStateToProps)(
-  withTranslation()(
-    ({ obj: project, index, rowKey, style, customData = {}, metrics, selectedColumns, t }) => {
-      const name = getName(project);
-      const requester = getRequester(project);
-      const {
-        ProjectLinkComponent,
-        actionsEnabled = true,
-        showMetrics,
-        showActions,
-        isColumnManagementEnabled = true,
-      } = customData;
-      const bytes = metrics?.memory?.[name];
-      const cores = metrics?.cpu?.[name];
-      const description = getDescription(project);
-      const labels = project.metadata.labels;
-      const columns = isColumnManagementEnabled
-        ? new Set(
-            selectedColumns?.get(projectColumnManagementID) ||
-              getProjectSelectedColumns({ showMetrics, showActions }),
-          )
-        : null;
-      return (
-        <TableRow id={project.metadata.uid} index={index} trKey={rowKey} style={style}>
-          <TableData className={namespaceColumnInfo.name.classes}>
-            {customData && ProjectLinkComponent ? (
-              <ProjectLinkComponent project={project} />
-            ) : (
-              <span className="co-resource-item">
-                <ResourceLink
-                  kind="Project"
-                  name={project.metadata.name}
-                  title={project.metadata.uid}
-                />
-              </span>
-            )}
-          </TableData>
-          <TableData
-            className={namespaceColumnInfo.displayName.classes}
-            columns={columns}
-            columnID={namespaceColumnInfo.displayName.id}
-          >
-            <span className="co-break-word co-line-clamp">
-              {getDisplayName(project) || (
-                <span className="text-muted">{t('namespace~No display name')}</span>
-              )}
+  withTranslation()(({ obj: project, index, rowKey, style, customData = {}, metrics, t }) => {
+    const [tableColumns] = useTableColumns(projectColumnManagementID);
```

Comment:
You need to check if the data has `loaded` otherwise this will flash the default columns first.
Same comment wherever this hook is used.